app/views.py

```diff
@@ -94,12 +94,15 @@ def post(self, request):
         request.session['name'] = name
 
         num_players = int(request.POST.get('num_players', 5))
+        has_mordred = request.POST.get('has_mordred', 'off') == 'on'
+        has_oberon =request.POST.get('has_oberon', 'off') == 'on'
+
+        if not gen_role_list(num_players, has_mordred, has_oberon):
+            messages.add_message(request, messages.ERROR, 'Too many spy roles, not enough spies.')
+            return redirect('create_game')
+
+        game = Game.games.create_game(num_players, has_mordred, has_oberon)
```

Comment:
might look nicer if we put validation into `Game.games.create_game` but I don't care much